pkg/cloudprovider/providers/azure/azure_cache.go

```diff
@@ -60,22 +82,56 @@ func (t *timedcache) GetOrCreate(key string, createFunc func() interface{}) (int
 		return nil, err
 	}
 	if exists {
-		return (entry.(*timedcacheEntry)).data, nil
+		return entry.(*cacheEntry), nil
 	}
 
-	if createFunc == nil {
-		return nil, nil
+	// Still not found, add new entry with nil data.
+	// Note the data will be filled later by getter.
+	newEntry := &cacheEntry{
+		key:  key,
+		data: nil,
 	}
-	created := createFunc()
-	t.store.Add(&timedcacheEntry{
+	t.store.Add(newEntry)
+	return newEntry, nil
+}
+
+// Get returns the requested item by key.
+func (t *timedCache) Get(key string) (interface{}, error) {
+	entry, err := t.getInternal(key)
+	if err != nil {
+		return nil, err
+	}
+
+	// Data is still not cached yet, cache it by getter.
+	if entry.data == nil {
+		entry.lock.Lock()
+		defer entry.lock.Unlock()
+
+		data, err := t.getter(key)
+		if err != nil {
+			return nil, err
+		}
+
+		entry.data = data
+	}
+
+	return entry.data, nil
+}
+
+// Update sets an item in the cache to its updated state.
+func (t *timedCache) Update(key string, data interface{}) error {
```

Comment:
Do we need this function? Suppose delete is enough, and the value should only be updated through getter

Reply:
Let's keep the cache function complete in case it be used in the future